pkg/controllers/csi/metadata/models.go

```diff
@@ -14,20 +14,20 @@ type TimeStampedModel struct {
 
 // TenantConfig holds info about a given configuration for a tenant.
 type TenantConfig struct {
-	CodeModule CodeModule `gorm:"foreignKey:Version;references:DowloadedCodeModuleVersion"`
+	CodeModule CodeModule `gorm:"foreignKey:DownloadedCodeModuleVersion;references:Version"`
 	TimeStampedModel
-	UID                        string `gorm:"primaryKey" json:"UID,omitempty"`
-	Name                       string `gorm:"not null"`
-	DowloadedCodeModuleVersion string
-	ConfigDirPath              string `gorm:"not null"`
-	TenantUUID                 string
-	OSMount                    OSMount `gorm:"foreignKey:TenantUUID;references:TenantUUID"`
-	MaxFailedMountAttempts     int64   `gorm:"default:10"`
+	UID                         string  `gorm:"primaryKey" json:"UID,omitempty"`
+	Name                        string  `gorm:"not null"`
+	DownloadedCodeModuleVersion string  `gorm:"not null"`
```

Comment:
would this `not null` constraint work in case of `HostMonitoring`?

Because in case of `HostMonitoring` we only do an `OSMount` , in which case we shouldn't download a CodeModule.

(I should have noticed this in the original schema PR 😬 )

Reply:
@0sewa0 without not null you can't set properly constraint, so we need remove link in this case.

Reply:
[but that is how it works](https://github.com/Dynatrace/dynatrace-operator/blob/efe236974140ed208f4dfda94ad0351599068770/pkg/controllers/csi/provisioner/controller.go#L125-L129)😞

Reply:
it's okey we can remove constraint, and just use as string, and for HostMonitoring it will be null.

Reply:
fixed in cbdf7da2